source/security/auth.txt

```diff
@@ -0,0 +1,160 @@
+credential by using the ``createCredential()`` static factory method:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-default
+   :end-before: end-default
+   :language: scala
+   :copyable:
+   :dedent:
+
+Or, you can use a connection string without explicitly specifying the
+authentication mechanism:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-default-connection-string
+   :end-before: end-default-connection-string
+   :language: scala
+   :copyable:
+   :dedent:
+
+For challenge and response mechanisms, using the default
+authentication mechanism is the recommended approach, as it makes
+upgrading from MongoDB 2.6 to MongoDB 3.0 more simple, even after
+upgrading the authentication schema. For MongoDB 4.0 users, using the
+default authentication mechanism is also recommended as the mechanisms are
+checked and the correct hashing algorithm is used.
+
+SCRAM-Based Mechanisms
+----------------------
+
+Salted Challenge-Response Authentication Mechanism (``SCRAM``) has been
+the default authentication mechanism for MongoDB since 3.0. ``SCRAM`` is
+based on the `IETF RFC 5802
+<https://datatracker.ietf.org/doc/html/rfc5802>`__ standard that defines
+best practices for implementation of challenge-response mechanisms for authenticating
+users with passwords.
+
+MongoDB 3.0 introduced support for ``SCRAM-SHA-1``, which uses the
+``SHA-1`` hashing function. MongoDB 4.0 introduced support for ``SCRAM-SHA-256`` which
+uses the ``SHA-256`` hashing function.
+
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+Using this mechanism requires MongoDB 4.0 and
+``featureCompatibilityVersion`` to be set to 4.0.
+
+To explicitly create a credential of type ``SCRAM-SHA-256``, use
+the ``createScramSha256Credential()`` method:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-scram-sha-256
+   :end-before: end-scram-sha-256
+   :language: scala
+   :copyable:
+   :dedent:
+
+Or, you can use a connection string that explicitly specifies
+``authMechanism=SCRAM-SHA-256``:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-scram-sha-256-connection-string
+   :end-before: end-scram-sha-256-connection-string
+   :language: scala
+   :copyable:
+   :dedent:
+
+SCRAM-SHA-1
+~~~~~~~~~~~
+
+To explicitly create a credential of type ``SCRAM-SHA-1``, use the
+``createScramSha1Credential()`` method:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-scram-sha-1
+   :end-before: end-scram-sha-1
+   :language: scala
+   :copyable:
+   :dedent:
+
+Or, you can use a connection string that explicitly specifies
+``authMechanism=SCRAM-SHA-1``:
+
+.. literalinclude:: /includes/security/auth.scala
+   :start-after: start-scram-sha-1-connection-string
+   :end-before: end-scram-sha-1-connection-string
+   :language: scala
+   :copyable:
+   :dedent:
+
+X.509
+-----
+
+With the ``X.509`` mechanism, MongoDB uses the X.509 certificate presented
```

Comment:
[s] If the first mention uses monospace, it makes sense for the other mentions to as well for consistency for example: 
```suggestion
With the ``X.509`` mechanism, MongoDB uses the ``X.509`` certificate presented
```
Or to remove from the first mention.

Reply:
Will remove it from the first mention since it seems to be the standard in other documentation.